### DISCUSSION TITLE
Pico Pi USB as Serial

### DISCUSSION THREAD

**Discussion by Dmkrd** [earlephilhower/arduino-pico#1855] [General]
Good afternoon I ask for advice...
I have a project on mega2560, it works with uart and sd card, and transmits data via serial to a program on a computer, the program collects data and analyzes it. due to the slow operation of the mega2560 with an sd card, I decided to switch to pico pi - I use Arduino IDE, I am very pleased with the speed of working with an sd card, but I encountered a problem transferring data to serial using usb pico pi to receive and transmit data..
Moreover, commands from my program pico pi accepts and processes them, namely the software does not accept data from pico pi... I thought that the problem was in the software, I tried a simple sketch using serial.print and a terminal program to analyze what was happening. As a result, despite the fact that the com port number and speed are correctly specified, the terminal does not accept data from the pico pi. So the question is: how can this problem be solved? I would like to use USB pico pi specifically for receiving and transmitting data using serial.print? if possible, please show me a couple of examples for a sample... thank you very much!

**Comment by Dmkrd** [comment]
Or use arduino nano as ttl converter ?)

**Reply by Dmkrd** [reply]
I tried it - it works!!))
serial1(0-1) - for work, serial2 (8-9) - for converter

**Reply by earlephilhower** [reply]
Look at https://github.com/earlephilhower/arduino-pico/blob/master/docs/serial.rst and the `Serial.ignoreFlowControl(true)` option.  Some Windows terminals don't set the proper virtual ready-to-receive flags and this call tells the Pico to just ignore the connection state and send anyway.

**Reply by Dmkrd** [reply]
Do you suggest using this method for USB connections or my method for - Serial2  to the converter?

**Reply by earlephilhower** [reply]
That's only on USB.  For UART, there's not normally a RTS/CTS pair wired up.

**Reply by Dmkrd** [reply]
Understood! I'm going to try now! Thank you!

**Comment by Dmkrd** [comment]
strange, after these settings pico pi is not recognized as a device in the system...

**Reply by earlephilhower** [reply]
Then you have some other issues, sorry.

**Comment by Dmkrd** [comment]
Can I ask you to add examples for working with serial? There is practically little devoted to this topic, if I was attentive. and there are a lot of questions...

**Reply by earlephilhower** [reply]
Just add it before you start sending things out/receiving in your `setup`.

**Comment by Dmkrd** [comment]
YES!! You are the best!!!
Serial.ignoreFlowControl(true) - this option helped!!! I made changes to my sketch and everything works great!! both the terminal and my software communicate perfectly with PICO PI via USB! Transmission and reception of data is clear!! Thanks a lot!!

**Comment by Dmkrd** [comment]
Today I started testing data transmission and realized that the most adequate speed is 115200, and the connection with my software is established adequately and data transmission is received without loss... is this speed the norm? I used to work at 38400 speed...
